app/models/user.rb

```diff
@@ -144,11 +144,9 @@ def self.ownership_request_notifiable_owners
     where(ownerships: { ownership_request_notifier: true })
   end
 
+  # Normalize email to ASCII-8BIT to avoid encoding issues.
   def self.normalize_email(email)
-    email.to_s.gsub(/\s+/, "")
-  rescue ArgumentError => e
-    Rails.error.report(e, handled: true)
-    ""
+    email.to_s.encode(Encoding::ASCII_8BIT, invalid: :replace, undef: :replace, replace: "").gsub(/\s+/, "")
```

Comment:
I'm a bit skeptical of silently replacing what a user entered... they could think they have 2 different email addressed that then "normalize" to the same address? since we're encoding to binary, could we instead keep the invalid/undef characters, and then where we do our email lookups, always bail out if the email isnt valid ascii?

Reply:
Should we maybe keep the default `�` inserted by `encode` normally? That should make the email invalid and yet keep it distinct (though it would be the same email no matter what invalid character was swapped in).

The alternative would be fully supporting RFC 6530 utf-8 emails. I'm not sure how hard that would be overall.

Reply:
I'm just reverting to simply rescuing encoding errors for now so we can avoid any behavior change. We can look at internationalized email in the future.